seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -311,28 +312,74 @@ def test_validate_anvil_vcf(self, mock_file_iter, mock_file_exist, mock_utils_lo
-                                           mock_send_email, mock_load_file, mock_api_logger, mock_datetime,
-                                           mock_airtable_logger, mock_utils_logger):
+
+PROJECT1_SAMPLES = ['HG00735', 'NA19675', 'NA19678', 'NA20870', 'HG00732', 'NA19675_1', 'NA20874', 'HG00733', 'HG00731']
+PROJECT2_SAMPLES = ['HG00735', 'NA19675', 'NA19678', 'NA20885']
+
+REQUEST_BODY_ADD_DATA = deepcopy(REQUEST_BODY)
+REQUEST_BODY_ADD_DATA['vcfSamples'] = PROJECT1_SAMPLES
+
+REQUEST_BODY_ADD_DATA2 = deepcopy(REQUEST_BODY)
+REQUEST_BODY_ADD_DATA2['vcfSamples'] = PROJECT2_SAMPLES
+
+PROJECT1_GUID = 'R0001_1kg'
+PROJECT2_GUID = 'R0003_test'
+ADD_DATA_UPDATED_ANVIL_VARIABLES = {
+    "key": "AnVIL_WES",
+    "value": json.dumps({
+        "active_projects": [PROJECT1_GUID],
+        "vcf_path": "gs://test_bucket/test_path.vcf",
+        "project_path": "gs://seqr-datasets/v02/GRCh37/AnVIL_WES/{guid}/v1".format(guid=PROJECT1_GUID),
+        "projects_to_run": [PROJECT1_GUID] })
+}
+ADD_DATA_UPDATE_DAG_TASKS_RESP = {
+            "tasks": [
+                {
+                    "task_id": "create_dataproc_cluster",
+                },
+                {
+                    "task_id": f"pyspark_compute_project_{PROJECT1_GUID}",
+                },
+                {
+                    "task_id": "pyspark_compute_variants_AnVIL_WES",
+                },
+                {
+                    "task_id": f"pyspark_export_project_{PROJECT1_GUID}",
+                },
+                {
+                    "task_id": "scale_dataproc_cluster",
+                },
+                {
+                    "task_id": f"skip_compute_project_subset_{PROJECT1_GUID}",
+                }
+                ],
+            "total_entries": 6
+        }
+
+AIRTABLE_TRACKING_URL = f'{MOCK_AIRTABLE_URL}/appUelDNM3BnWaR7M/AnVIL%20Seqr%20Loading%20Requests%20Tracking'
+
+
+@mock.patch('seqr.views.apis.anvil_workspace_api.id_token.fetch_id_token', lambda *args: MOCK_TOKEN)
+@mock.patch('seqr.views.utils.airtable_utils.AIRTABLE_API_KEY', MOCK_AIRTABLE_KEY)
+@mock.patch('seqr.views.utils.airtable_utils.AIRTABLE_URL', MOCK_AIRTABLE_URL)
+@mock.patch('seqr.views.apis.anvil_workspace_api.AIRFLOW_WEBSERVER_URL', MOCK_AIRFLOW_URL)
+@mock.patch('seqr.views.apis.anvil_workspace_api.BASE_URL', 'http://testserver/')
+@mock.patch('seqr.views.utils.permissions_utils.logger')
+@mock.patch('seqr.views.utils.airtable_utils.logger')
+@mock.patch('seqr.views.apis.anvil_workspace_api.load_uploaded_file')
+@mock.patch('seqr.views.apis.anvil_workspace_api.safe_post_to_slack')
+@mock.patch('seqr.views.apis.anvil_workspace_api.mv_file_to_gs')
+@mock.patch('seqr.views.apis.anvil_workspace_api.tempfile.NamedTemporaryFile')
+@mock.patch('seqr.views.apis.anvil_workspace_api.logger')
+@mock.patch('seqr.views.apis.anvil_workspace_api.datetime')
+@mock.patch('seqr.views.apis.anvil_workspace_api.send_html_email')
```

Comment:
we end up passing a lot of these mocks arounds as arguments to various functions, which works but is hard to read/maintain. I think it would be better to define the patchers in the `setUp` method similar to what we do for `AnvilAuthenticationTestCase` so that the mocks are actually class properties. For example
```
        patcher = mock.patch('seqr.views.apis.anvil_workspace_api.send_html_email'')
        self.mock_send_email = patcher.start()
        self.addCleanup(patcher.stop)
```
Then `self._assert_sent_slack_message` would no longer take `mock_send_email` as an argument but would instead access `self.mock_send_email` from inside the function call